packages/default/scss/button/_layout.scss

```diff
@@ -211,13 +208,18 @@
     }
 
 
-    // Flat button
-    .k-button.k-flat {
-        border-color: transparent;
+    // Flat button and bare
+    .k-button.k-flat,
+    .k-button.k-bare {
+        border-color: transparent !important; // sass-lint:disable-line no-important
         color: inherit;
-        background-color: transparent;
-        background-image: none;
-        box-shadow: none;
+        background: none !important; // sass-lint:disable-line no-important
+        box-shadow: none !important; // sass-lint:disable-line no-important
+        transition: color .2s ease-in-out;
+
+        &::before {
+            display: block;
+        }
```

Comment:
Assuming we always want flat / bare buttons to behave like such, I think it's safe to `!important` rules.

Reply:
Are these required? While this is okay for most scenarios, it will be very hard to override, if we or our customers ever need to.

Reply:
IMO, yes. If we agree that a flat button is one that behaves a certain way, then that's the way to go.

Alternatively, we could move the styles in `button/theme` for each theme. It will not change the output, per say, but will allow for a developer, be it us or a customer, to tweak each theme individually, should there is a need.

Reply:
I still feel a bit uneasy for using `!important` on buttons, but let's see how far it goes. If this obstructs functionality, we can split this to the theme files.